applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -572,44 +612,42 @@ static int update_sink_stream_qos(struct le_audio_headset *headset, uint32_t pre
 {
 	int ret;
 
-	if (headset->sink_stream.ep == NULL) {
+	if (headset->sink_stream.bap_stream.ep == NULL) {
 		return -ESRCH;
 	}
 
-	if (headset->sink_stream.qos == NULL) {
-		LOG_WRN("No QoS found for %p", &headset->sink_stream);
+	if (headset->sink_stream.bap_stream.qos == NULL) {
+		LOG_WRN("No QoS found for %p", &headset->sink_stream.bap_stream);
 		return -ENXIO;
 	}
 
-	if (headset->sink_stream.qos->pd != pres_delay_us) {
-		if (playing_state &&
-		    le_audio_ep_state_check(headset->sink_stream.ep, BT_BAP_EP_STATE_STREAMING)) {
+	if (headset->sink_stream.bap_stream.qos->pd != pres_delay_us) {
+		if (playing_state && le_audio_ep_state_check(headset->sink_stream.bap_stream.ep,
+							     BT_BAP_EP_STATE_STREAMING)) {
 			LOG_DBG("Update streaming %s headset, connection %p, stream %p",
-				headset->ch_name, &headset->headset_conn, &headset->sink_stream);
+				headset->ch_name, &headset->headset_conn,
+				&headset->sink_stream.bap_stream);
 
 			headset->qos_reconfigure = true;
 			headset->reconfigure_pd = pres_delay_us;
 
-			ret = bt_bap_stream_disable(&headset->sink_stream);
+			struct bt_cap_unicast_audio_stop_param param;
+			struct bt_cap_stream *streams[1];
+			streams[0] = &headset->sink_stream;
+
+			param.streams = streams;
+			param.count = 1;
+			param.type = BT_CAP_SET_TYPE_AD_HOC;
+
+			ret = bt_cap_initiator_unicast_audio_stop(&param);
 			if (ret) {
-				LOG_ERR("Unable to disable stream: %d", ret);
+				LOG_ERR("Unable to stop stream: %d", ret);
```

Comment:
Since all streams are stopped, I recommend we change this string to something more general. "Failed to stop streams", "CAP audio stop failed" ...

Reply:
In this case only one stream is stopped though, hence the `param.count=1`

Reply:
Checking source as well now, so changed to `"Failed to stop streams: %d", ret`